code/modules/roguetown/roguemachine/merchant.dm

```diff
@@ -122,6 +122,7 @@
 	var/budget = 0
 	var/upgrade_flags
 	var/current_cat = "1"
+	var/hidden_key = TRUE
```

Comment:
```suggestion
	//Whether or not the hidden key is still present- Shophands can take it.
	var/hidden_key_present = TRUE
```
There's better ways to put this, but yeah.

Reply:
Alright, changed the name around eveywhere